src/targets/shell/curl/client.ts

```diff
@@ -0,0 +1,134 @@
+/**
+ * @description
+ * HTTP code snippet generator for the Shell using cURL.
+ *
+ * @author
+ * @AhmadNassri
+ *
+ * for any questions or issues regarding the generated code snippet, please open an issue mentioning the author.
+ */
+
+import { CodeBuilder } from '../../../helpers/code-builder';
+import { getHeaderName } from '../../../helpers/headers';
+import { quote } from '../../../helpers/shell';
+import { Client } from '../../targets';
+
+export interface CurlOptions {
+  short?: boolean;
+  binary?: boolean;
+  globOff?: boolean;
+}
+
+export const curl: Client<CurlOptions> = {
+  info: {
+    key: 'curl',
+    title: 'cURL',
+    link: 'http://curl.haxx.se/',
+    description: 'cURL is a command line tool and library for transferring data with URL syntax',
+  },
+  convert: ({ fullUrl, method, httpVersion, headersObj, allHeaders, postData }, options) => {
+    const opts = {
+      indent: '  ',
+      short: false,
+      binary: false,
+      globOff: false,
+      ...options,
+    };
+    const { push, join } = new CodeBuilder({
+      indent: opts.indent,
+      // @ts-expect-error SEEMS LEGIT
+      join: opts.indent !== false ? ` \\\n${opts.indent}` : ' ',
```

Comment:
The typing error on this one is definitely strange. Could maybe do an `opts.indent.length` check instead to resolve it?

Reply:
I resloved it with type-fest's `Merge`: https://github.com/sindresorhus/type-fest/blob/main/source/merge.d.ts

see 7ea5e5d16bc4cad95ab9970a1b210b9a48f6c5a7